services/storage/src/simcore_service_storage/db_file_meta_data.py

```diff
@@ -74,33 +74,47 @@ async def list_filter_with_partial_file_id(
     partial_file_id: str | None,
     sha256_checksum: SHA256Str | None,
     only_files: bool,
+    limit: int | None = None,
+    offset: int | None = None,
 ) -> list[FileMetaDataAtDB]:
-    stmt = sa.select(file_meta_data).where(
-        (
-            (file_meta_data.c.user_id == f"{user_id}")
-            | file_meta_data.c.project_id.in_(f"{pid}" for pid in project_ids)
-        )
-        & (
-            file_meta_data.c.file_id.startswith(file_id_prefix)
-            if file_id_prefix
-            else True
-        )
-        & (
-            file_meta_data.c.file_id.ilike(f"%{partial_file_id}%")
-            if partial_file_id
-            else True
-        )
-        & (
-            file_meta_data.c.is_directory.is_(False)  # noqa FBT003
-            if only_files
-            else True
-        )
-        & (
-            file_meta_data.c.sha256_checksum == f"{sha256_checksum}"
-            if sha256_checksum
-            else True
+    stmt = (
+        sa.select(file_meta_data).where(
+            (
+                (file_meta_data.c.user_id == f"{user_id}")
+                | file_meta_data.c.project_id.in_(f"{pid}" for pid in project_ids)
+            )
+            & (
+                file_meta_data.c.file_id.startswith(file_id_prefix)
+                if file_id_prefix
+                else True
+            )
+            & (
+                file_meta_data.c.file_id.ilike(f"%{partial_file_id}%")
+                if partial_file_id
+                else True
+            )
+            & (
+                file_meta_data.c.is_directory.is_(False)  # noqa FBT003
+                if only_files
+                else True
+            )
+            & (
+                file_meta_data.c.sha256_checksum == f"{sha256_checksum}"
+                if sha256_checksum
+                else True
+            )
         )
+        # oldest first
+        .order_by(file_meta_data.c.created_at.asc())
```

Comment:
FYI if you would want to optimize in the future also index on `created_at` might potentially help, but it would be good to analyse this query with `explanation select ...` to be sure.

Reply:
thx! Very very valid point. I will check and add if applies.

Reply:
I suggest to try and profile this first to see what the overhead is by adding this with the current db configuration. I worry that this sorting could bring a large overhead (defeating the purpose of the PR). I understand why you do it, but would be good to double check.

Reply:
I thought the same. I am on it.

Reply:
peer reviewed with @matusdrobuliak66 .  Checked performance and left indexing for later. No need for a change now.